### DISCUSSION TITLE
How to return timestamps in different format?

### DISCUSSION THREAD

**Discussion by c6s0** [snakers4/silero-vad#184] [Q&A] [answered]
How can i return the timestamps in `h:m:s` format and not in `s.ms` ?

I guess you should be able to do it somewhere here but i can't find the specified format? 

    if return_seconds:
        for speech_dict in speeches:
            speech_dict['start'] = round(speech_dict['start'] / sampling_rate, 1)
            speech_dict['end'] = round(speech_dict['end'] / sampling_rate, 1)
    elif step > 1:
        for speech_dict in speeches:
            speech_dict['start'] *= step
            speech_dict['end'] *= step

**Comment by ryanlong1004** [comment, marked as answer]
I'm guessing speech_dict start and end times are epoch timestamps,  correct?

**Reply by snakers4** [reply]
> I'm guessing speech_dict start and end times are epoch timestamps, correct?

Not sure what you mean by epochs in this context, but these are indeed seconds.

> How can i return the timestamps in h:m:s format and not in s.ms ?
> I guess you should be able to do it somewhere here but i can't find the specified format?

There is no format.
These are just seconds with some precision.
Typically less than 50-100ms is not practical in a non-streaming fashion.
If you need another format - you can just override this function and / or replace with yours.

**Comment by dementedAngel** [comment]
Okay, so I gather that the timestamps are in s.ms format. how many digits are given for the ms?  I'm trying to figure out what my data mean!

**Reply by snakers4** [reply]
https://github.com/snakers4/silero-vad/blob/262bcb4b4043e29e19c0b578f7d4249345bba55e/utils_vad.py#L170

**Reply by snakers4** [reply]
It is either seconds or samples

**Comment by dementedAngel** [comment]
I figured out that if I divide by 10000, I'm getting about 480 'seconds' in a 5-minute segment. So, to get the real time I need to divide again by 1.6.

**Comment by dementedAngel** [comment]
So yes, it's samples.

**Reply by rjac-ml** [reply]
yes the format that returns is in sample rate. that is very useful given that the imput wav are sample rate too. So if you want to chunk/parse that audio wav then is very straight forward with this values
